mimic/session.py

```diff
@@ -120,9 +124,9 @@ def session_for_token(self, token, tenant_id=None):
             if tenant_id is not None and s.tenant_id != tenant_id:
                 raise NonMatchingTenantError(session=s,
                                              desired_tenant=tenant_id)
-            return s
-
-        return self._new_session(token=token, tenant_id=tenant_id)
+        else:
+            s = self._new_session(token=token, tenant_id=tenant_id)
+        return s, self._impersonated_token_to_impersonator_session.get(token)
```

Comment:
I don't really like changing the signature of `_new_session` (and thus all of its callers) to a 2-tuple; tuples are much less self-describing than (characteristic) structures with attributes, so unless it's really obvious which is the first thing and which is the second thing in the return value, it's much better to have attributes with names that at least imply what they are.